query/device_allocator.go

```diff
@@ -100,20 +102,22 @@ type deviceAllocator interface {
 	deviceAllocate(bytes, device int) devicePointer
 	// deviceFree frees the specified memory from the device.
 	deviceFree(dp devicePointer)
+	// getAllocatedMemory returns allocated memory for a specific device.
+	getAllocatedMemory(device int) int64
 }
 
 var da deviceAllocator
 var deviceAllocatorOnce sync.Once
 
-// getDeviceAllocator returns singleton deviceAllocatorImpl instance.
+// getDeviceAllocator returns singleton memoryTrackingDeviceAllocatorImpl instance.
```

Comment:
DeviceAllocator? now there are multiple implementations

Reply:
yep, it's due to the refactor I will rename them